doc/source/packaging/structure.rst

```diff
@@ -0,0 +1,306 @@
+PyAnsys projects use the `pytest`_ testing framework.
+
+A good practice is to emulate the structure of the ``src/ansys/product/library``
+directory although it is not always necessary.
+
+.. include:: diag/tests_structure_diag.rst
+
+Notice the usage of ``tests_*/`` when creating new directories inside the
+``tests/`` one. On the other hand, unit testing files are named using the
+``test_*.py`` prefix. This is the preferred way of naming directories and files
+inside the ``tests/`` directory.
+
+
+The ``LICENSE`` File
+--------------------
+
+The ``LICENSE`` file provides the legal framework for the software. The
+recommended license for `PyAnsys`_ projects is `MIT License`_. A template for
+this license is provided below:
+
+.. include:: code/license_mit_code.rst
+
+.. note::
+
+   Just because a software does not ship with a LICENSE file, it does not mean
+   it is free or open source. If you need to use unlicensed software, contact
+   its development team so they can provide you with the right license.
+
+
+The ``README.rst`` File
+-----------------------
+
+Each PyAnsys library should have a ``README.rst`` file in the root directory.
+
+The preferred format of this file is `reStructuredText Markup Syntax`_ or a
+although `Markdown Syntax`_ can be used too.  While Markdown syntax has better
+GitHub support, text in RST files can be reused within Sphinx documentation.
+This avoids duplication between the ``README.rst`` and the main ``index.rst`` in
+the ``doc/source/`` directory.
+
+The ``README.rst`` file should at the minimum contain these elements:
+
+- PyAnsys library title
+- General description
+- Installation directions (via ``pip install`` and ``git clone``)
+- Basic usage
+- Links to the full documentation
+
+The ``README.rst`` file is also reused within the project file metadata. It is
+usually included in the ``long-description`` field.
+
+
+The ``pyproject.toml`` File
+---------------------------
+
+The `PEP 518`_ introduced the usage of a new project file called
+``pyproject.toml``. More information about this file can be found in the
+REF-TO-BUILD-SYSTEMS-TOML section.
+
+This file is mandatory as it  allows ``pip`` to resolve the requirements for
+building the library. The following tabs expose the ``[build-system]`` section
+for some of the most popular build-system backend tools in the Python ecosystem:
+
+.. include:: code/pyproject_code.rst
+
+
+The ``setup.py`` File
+---------------------
+
+For a long time, the ``setup.py`` has been the usual way of building and
+distributing Python libraries. As opposite to the static ``pyproject.toml``
+file, ``setup.py`` is Python script. This means that Python code is interpreted
+when building the library. This allows to customize the building process but
+also may introduced security issues.
+
+.. warning::
+
+   The ``setup.py`` is only compatible with `setuptools`_. Consider using a
+   ``pyproject.toml`` file instead.
+
```

Comment:
@jorgepiloto Warning are generally reserved for important issues like loss of data. It seems that this warning should be a note instead.